qiskit/compiler/transpiler.py

```diff
@@ -607,8 +619,14 @@ def _create_faulty_qubits_map(backend):
 def _parse_basis_gates(basis_gates, backend, circuits):
     # try getting basis_gates from user, else backend
     if basis_gates is None:
-        if getattr(backend, "configuration", None):
-            basis_gates = getattr(backend.configuration(), "basis_gates", None)
+        backend_version = getattr(backend, "version", 0)
+        if not isinstance(backend_version, int):
+            backend_version = 0
+        if backend_version <= 1:
+            if getattr(backend, "configuration", None):
+                basis_gates = getattr(backend.configuration(), "basis_gates", None)
+        else:
+            basis_gates = backend.instructions_names
```

Comment:
Since we're dropping the expectation that all gates are available on all qubits/edges, `backend.instructions_names` might not always be a drop-in replacement for `basis_gates`  (e.g. it's not clear that a pass expecting `basis_gates` would be able to do anything helpful if the device has an non-homogeneous gate set). Would it make sense to have a similar compatibility shim on the backend `basis_gates` that warns if the device doesn't support all gates on all qubits?

Reply:
Yeah, that's probably a good idea. I think that's a reasonable way to handle this until we can update the transpiler to start working natively with a `Target`. I'll add a similar warning as to what's emitted when you ask for a coupling map if there are >2q gates in the target.